src/socket.cpp

```diff
@@ -40,25 +50,46 @@ void CSocket::Init ( const quint16 iPortNumber, const quint16 iQosNumber, const
 #endif
 
     // create the UDP socket
-    UdpSocket = socket ( AF_INET, SOCK_DGRAM, 0 );
-    //
-    const char tos = (char) iQosNumber; // Quality of Service
-    setsockopt ( UdpSocket, IPPROTO_IP, IP_TOS, &tos, sizeof ( tos ) );
+    UdpSocket = socket ( AF_INET6, SOCK_DGRAM, 0 );
+    if ( UdpSocket == -1 )
+    {
+        // IPv6 not available; fall back to IPv4
+        bUseIPV6 = false;
+
+        UdpSocket = socket ( AF_INET, SOCK_DGRAM, 0 );
+        //
+        const char tos = (char) iQosNumber; // Quality of Service
+        setsockopt ( UdpSocket, IPPROTO_IP, IP_TOS, &tos, sizeof ( tos ) );
+    }
+    else
+    {
+        // The IPV6_V6ONLY socket option must be false in order for the socket to listen on both protocols.
+        // On Linux it's false by default on most (all?) distros, but on Windows it is true by default
+        const uint8_t no = 0;
+        setsockopt( UdpSocket, IPPROTO_IPV6, IPV6_V6ONLY, (const char *)&no, sizeof( no ) );
+        //
+        const char tos = (char) iQosNumber; // Quality of Service
+        setsockopt ( UdpSocket, IPPROTO_IPV6, IPV6_TCLASS, &tos, sizeof ( tos ) );
+    }
 
     // allocate memory for network receive and send buffer in samples
     vecbyRecBuf.Init ( MAX_SIZE_BYTES_NETW_BUF );
 
     // preinitialize socket in address (only the port number is missing)
     sockaddr_in UdpSocketInAddr;
-    UdpSocketInAddr.sin_family = AF_INET;
-    if ( strServerBindIP.isEmpty() )
-    {
-        UdpSocketInAddr.sin_addr.s_addr = INADDR_ANY;
-    }
-    else
+    UdpSocketInAddr.sin_family      = AF_INET;
+    UdpSocketInAddr.sin_addr.s_addr = INADDR_ANY;
+
+    struct sockaddr_in6 UdpSocketIn6Addr;
```

Comment:
The rest of the network code avoids writing `struct` when it's not necessary in C++ (`sockaddr`, `sockaddr_in`); can this and the sizeof below just be `sockaddr_in6`?

Reply:
Probably best to review just the differences between the branch point and the final commit.